spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -157,6 +158,16 @@ function verifyChannelUpgrade(
 }
 ```
 
+#### CounterpartyNextSequenceSend Path
+
+The chain must store the counterparty's next sequence send on `startFlushUpgradeHandshake`. This will be stored in the `counterpartyNextSequenceSend` path on the private store.
+
```

Comment:
is it possible to access the counterparty upgrade information for example `OnChanUpgradeTry`? I think the full counterparty upgrade is not set until `WriteUpgradeAck` / `WriteUpgradeConfirm`, at this point we only have the upgrade fields

Reply:
You're right, I think the spec is missing the call to `privateStore.set(counterpartyNext...` with `counterpartyUpgrade.nextPacketSend` that should happen in `chanUpgradeAck` and `chanUpgradeConfirm`. I mentioned that to @AdityaSripal this morning and he was going to look into it.

Reply:
You're correct. Thanks, added the counterparty set now to ack and confirm